server/storage/mvcc/revision.go

```diff
@@ -14,48 +14,84 @@
 
 package mvcc
 
-import "encoding/binary"
+import (
+	"encoding/binary"
 
-// revBytesLen is the byte length of a normal revision.
-// First 8 bytes is the revision.main in big-endian format. The 9th byte
-// is a '_'. The last 8 bytes is the revision.sub in big-endian format.
-const revBytesLen = 8 + 1 + 8
+	"go.uber.org/zap"
+)
 
-// A revision indicates modification of the key-value space.
-// The set of changes that share same main revision changes the key-value space atomically.
-type revision struct {
-	// main is the main revision of a set of changes that happen atomically.
-	main int64
+const (
+	// revBytesLen is the byte length of a normal revision.
+	// First 8 bytes is the revision.main in big-endian format. The 9th byte
+	// is a '_'. The last 8 bytes is the revision.sub in big-endian format.
+	revBytesLen = 8 + 1 + 8
+	// markedRevBytesLen is the byte length of marked revision.
+	// The first `revBytesLen` bytes represents a normal revision. The last
+	// one byte is the mark.
+	markedRevBytesLen      = revBytesLen + 1
+	markBytePosition       = markedRevBytesLen - 1
+	markTombstone     byte = 't'
+)
 
-	// sub is the sub revision of a change in a set of changes that happen
+// RevisionTombstone indicates modification of the key-value space.
```

Comment:
RevisionTombstone as a name describes that it holds, but it's hard to understand what's its purpose. 
The etcd MVCC managed they bbolt key bucket, which uses RevisionTombstone as a key. Have you thought about naming it based on it's purpose? For example `mvcc.Key` or `mvcc.BucketKey`. This